code/modules/network/network.dm

```diff
@@ -0,0 +1,190 @@
+/*
+	Upgraded devices, such as borgs, the AI, or "Hacktools" have several notable features:
+	they are able to see stealthed networks.
+	they are able to pass "-b" as an argument in place of an encryption key to make an attempt at brute-forcing the network. This will take some time, and may trigger the networks security system, but will eventually get you in.
+*/
+
+var/global/list/networks_by_id = list()
+var/global/list/datum/network/networks_by_wide = list()
+var/global/list/active_network_ids = list()
+
+proc/clean_network_command(command = "")
+	var/list/butchered = explode_text(command, " -") // allows for arguments like {text1 = "hello "} {text2 = "world"} to be caused when executing something like {print -text1 = "hello " -text2 = "world"}
+	command = butchered[1]
+	var/list/arguments = list()
+	for(var/i=2, i<=butchered.len, i++)
+		arguments[i-1] = butchered[i]
+	return list(command, arguments)
+
+proc/getuniqueid()
```

Comment:
>splittext proc
See also: findtext proc jointext proc nonspantext proc spantext proc Regular expressions Format: splittext(Text,Delimiter,Start=1,End=0,include_delimiters=0) Returns: A list of text strings split by the delimiter given. Args: Text: The text string to search. Delimiter: A text string that will be used as the separator between items, OR a regular expression (regex) used to find splits Start: The text character position in Text in which to begin. End: The text character position in Text immediately following the last character to include in the splitting process. include_delimiters: True if any delimiters found should be included in the result. 

You need to be on 510, we're not 509 compat anymore.